modin_spreadsheet/tests/test_grid.py

```diff
@@ -1084,3 +1084,66 @@ def test_mixed_type_sort():
     expected_order_1 = [1, 2, 4, 5, "hello"]
     sorted_order_1 = list(spreadsheet.get_changed_df()[1])
     assert expected_order_1 == sorted_order_1
+
+
+def test_column_reorder():
+    spreadsheet = SpreadsheetWidget(df=create_df())
+
+    spreadsheet._handle_view_msg_helper(
+        {"column_names": ["Date", "C", "D", "E", "F", "A"], "type": "reorder_columns"}
+    )
+
+    expected_columns = ["Date", "C", "D", "E", "F", "A"]
+    actual_columns = list(spreadsheet.get_changed_df().columns)
+    assert expected_columns == actual_columns
+
+    spreadsheet._handle_view_msg_helper(
+        {"column_names": ["Date", "C", "F", "D", "E", "A"], "type": "reorder_columns"}
+    )
+
+    expected_columns = ["Date", "C", "F", "D", "E", "A"]
+    actual_columns = list(spreadsheet.get_changed_df().columns)
+    assert expected_columns == actual_columns
+
+
+def test_column_reorder_multi():
+    df = pd.DataFrame(
+        {
+            "A": [1, 2, 3, 4],
+            "Date": pd.Timestamp("20130102"),
+            "C": pd.Series(1, index=list(range(4)), dtype="float32"),
+            "D": np.array([3] * 4, dtype="int32"),
+            "E": pd.Categorical(["test", "train", "foo", "bar"]),
+            "F": ["foo", "bar", "buzz", "fox"],
+        }
+    )
+
+    spreadsheet = SpreadsheetWidget(df=df)
+
+    spreadsheet._handle_view_msg_helper(
+        {"column_names": ["Date", "C", "A", "D", "E", "F"], "type": "reorder_columns"}
+    )
+
+    expected_columns = ["Date", "C", "A", "D", "E", "F"]
+    actual_columns = list(spreadsheet.get_changed_df().columns)
+    assert expected_columns == actual_columns
+
+    spreadsheet._handle_view_msg_helper(
+        {"type": "change_sort", "sort_field": "A", "sort_ascending": False}
+    )
+
+    expected_order = [4, 3, 2, 1]
+    sorted_order = list(spreadsheet.get_changed_df()["A"])
+    assert expected_order == sorted_order
+
+    spreadsheet._handle_view_msg_helper(
+        {
+            "row_index": 0,
```

Comment:
`row_index` and `unfiltered_index` should technically be the same
```suggestion
            "row_index": 3,
```

Reply:
The comment is wrong, they don't have to be the same. The suggestion is still correct.